### ISSUE TITLE
sensu-client incorrectly reports a fatal error indicating subscriptions is not an array

### ISSUE THREAD

**Issue by cwjohnston** [sensu/sensu-settings#47]
On a system with Sensu 0.26.3 and sensu-settings 9.2.0, starting the sensu-client with default configuration fails with the following fatal error:

```
{"timestamp":"2016-10-05T17:51:54.894746+0000","level":"fatal","message":"client subscriptions must be an array","object":{"name":"default-debian-78","address":"10.0.2.15","subscriptions":["client:default-debian-78"]}}
```
The output indicates that subscriptions configuration is an array, and it is populated with the automatic `client:` subscription. This seems like a bug.

**Comment by cwjohnston** [comment]
As a workaround, try providing an array as the value for `subscriptions` under the `client` scope, e.g.:

```
{ "client": { "subscriptions": [] } }
```

Doesn't have to be empty, just needs to be an array.

**Comment by eladishay** [comment]
@cwjohnston I've encountered the same issue, here is my client.json under C:\opt\sensu\conf.d\

{
	"client": {
		"name": "windows",
		"address": "127.0.0.1",
		"environment": "development",
		"subscriptions": [
			"dev",
			"windows"
		]
	}
}

And the error log: 

{"timestamp":"2016-10-09T22:22:01.735198+0300","level":"warn","message":"applied sensu client overrides","client":{"name":"ELADISHAYAF76","address":"10.211.55.5","subscriptions":["client:ELADISHAYAF76"]}}
{"timestamp":"2016-10-09T22:22:01.735198+0300","level":"fatal","message":"client subscriptions must be an array","object":{"name":"ELADISHAYAF76","address":"10.211.55.5","subscriptions":["client:ELADISHAYAF76"]}}
{"timestamp":"2016-10-09T22:22:01.735198+0300","level":"fatal","message":"SENSU NOT RUNNING!"}

It seems like it completely ignores my client.json file. Any idea? Tried it under Windows server 2008 r2 and windows 10.

10x a lot.

**Comment by cwjohnston** [comment]
@eladishay can you please check your logs to ensure the config from `C:\opt\sensu\conf.d\client.json` is being read in successfully?

In my local testing I'm seeing that defining the client subscription array, empty or with one or more elements, in any valid loaded config seems to work.

**Comment by eladishay** [comment]
@cwjohnston Yesss all good now! No idea how but I've missed the -c and -d arguments... 
Thanks again for your help!!

  <arguments>C:\opt\sensu\embedded\bin\sensu-client -c C:\opt\sensu\config.json -d C:\opt\sensu\conf.d -l C:\opt\sensu\sensu-client.log</arguments>

**Comment by cwjohnston** [comment]
👍

**Comment by portertech** [comment]
Moving overrides before validation should fix this issue, as it will set client subscriptions appropriately.

**Comment by portertech** [comment]
https://github.com/sensu/sensu-settings/blob/master/lib/sensu/settings.rb#L27-L30

**Comment by kwright5150** [comment]
I have tried this on a Windows 7 client and I am receiving the same log information. I tried just the bare minimum config found here: https://sensuapp.org/docs/0.26/platforms/sensu-on-microsoft-windows.html

Same result:

"applied sensu client overrides"
"client subscriptions must be an array"
"SENSU NOT RUNNING!"

I am running the latest client version.

**Comment by cwjohnston** [comment]
@kwright5150 sorry to report that the download link on sensuapp.org is out of date. The most recent windows build is here: https://sensu.global.ssl.fastly.net/msi/sensu-0.26.5-1.msi

I believe this Windows build includes the fix described in #48

**Comment by kwright5150** [comment]
Awesome, thank you!
